components/views/dropdown-menu-view.js

```diff
@@ -0,0 +1,231 @@
+define([
+        'backbone',
+        'jquery',
+        'underscore',
+        'text!../templates/dropdown.underscore'
+    ],
+    function( Backbone, $, _, DropdownTpl ) {
+        'use strict';
+
+        return Backbone.View.extend({
+            tpl: _.template( DropdownTpl ),
+
+            events: {
+                'click .js-dropdown-button': 'clickOpenDropdown',
+                'click a': 'analyticsLinkClick',
+                'keydown': 'viewKeypress'
+            },
+
+            menu: '.dropdown-menu',
+
+            key: {
+                tab: 9,
+                enter: 13,
+                esc: 27,
+                up: 38,
+                down: 40
+            },
+
+            initialize: function( options ) {
+                this.$parent = $(options.parent);
+                this.render();
+            },
+
+            className: function() {
+                return this.options.className;
+            },
+
+            render: function() {
+                this.$el.html( this.tpl( this.model.toJSON() ) );
+                this.$parent.replaceWith( this.$el );
+                this.postRender();
+
+                return this;
+            },
+
+            postRender: function() {
+                this.$menu = this.$el.find('.dropdown-menu');
+                this.$page = $(document);
+                this.$dropdownButton = $('.js-dropdown-button');
+                this.listenForPageKeypress();
+            },
+
+            analyticsLinkClick: function( event ) {
+                var $link = $(event.target),
+                    label = $link.hasClass('user-title') ? 'Dashboard' : $link.html().trim();
+
+                /**
+                 *  Add your own analytics tracking here
+                 *  for example:
+                 */
+                window.analytics.track( 'user_dropdown.clicked', {
```

Comment:
For tests to run I kept this in as a default (otherwise it wouldn't be easy to determine whether analytics tracking was working).